metricbeat/module/vsphere/host/host.go

```diff
@@ -92,83 +111,187 @@ func (m *MetricSet) Fetch(ctx context.Context, reporter mb.ReporterV2) error {
+		"disk.write.average",
+		"net.transmitted.average",
+		"net.received.average",
+		"net.usage.average",
+		"net.packetsTx.summation",
+		"net.packetsRx.summation",
+		"net.errorsTx.summation",
+		"net.errorsRx.summation",
+		"net.multicastTx.summation",
+		"net.multicastRx.summation",
+		"net.droppedTx.summation",
+		"net.droppedRx.summation",
+	}
+
+	// Define reference of structure
+	var metricsVar PerformanceMetrics
+
+	// Map metric names to structure fields
+	metricMap := map[string]*int64{
+		"disk.capacity.usage.average": &metricsVar.DiskCapacityUsage,
+		"disk.deviceLatency.average":  &metricsVar.DiskDeviceLatency,
+		"disk.maxTotalLatency.latest": &metricsVar.DiskMaxTotalLatency,
+		"net.usage.average":           &metricsVar.NetUsage,
+		"disk.usage.average":          &metricsVar.DiskUsage,
+		"disk.read.average":           &metricsVar.DiskRead,
+		"disk.write.average":          &metricsVar.DiskWrite,
+		"net.transmitted.average":     &metricsVar.NetTransmitted,
+		"net.received.average":        &metricsVar.NetReceived,
+		"net.packetsTx.summation":     &metricsVar.NetPacketTransmitted,
+		"net.packetsRx.summation":     &metricsVar.NetPacketReceived,
+		"net.errorsTx.summation":      &metricsVar.NetErrorsTransmitted,
+		"net.errorsRx.summation":      &metricsVar.NetErrorsReceived,
+		"net.multicastTx.summation":   &metricsVar.NetMulticastTransmitted,
+		"net.multicastRx.summation":   &metricsVar.NetMulticastReceived,
+		"net.droppedTx.summation":     &metricsVar.NetDroppedTransmitted,
+		"net.droppedRx.summation":     &metricsVar.NetDroppedReceived,
+	}
+
+	// Retrieve only the required metrics
+	requiredMetrics := make(map[string]*types.PerfCounterInfo)
+
+	for _, name := range metricNames {
+		metric, exists := metrics[name]
+		if !exists {
+			m.Logger().Warnf("Metric %s not found", name)
+			continue
+		}
+		requiredMetrics[name] = metric
+	}
+
+	metricIDs := make([]types.PerfMetricId, 0, len(requiredMetrics))
+	for _, metric := range requiredMetrics {
+		metricIDs = append(metricIDs, types.PerfMetricId{
+			CounterId: metric.Key,
+		})
+	}
+
+	pc := property.DefaultCollector(c)
 	for _, hs := range hst {
+		datastoreRefs := make([]types.ManagedObjectReference, 0, len(hs.Datastore))
+		for _, obj := range hs.Datastore {
+			if obj.Type == "Datastore" {
+				datastoreRefs = append(datastoreRefs, obj)
+			}
+		}
 
-		event := mapstr.M{}
+		var datastores []mo.Datastore
+		if len(datastoreRefs) > 0 {
+			if err := pc.Retrieve(ctx, datastoreRefs, []string{"name"}, &datastores); err != nil {
+				m.Logger().Errorf("Failed to retrieve datastore from host: %v", err)
+				continue
+			}
+		}
 
-		event["name"] = hs.Summary.Config.Name
-		event.Put("cpu.used.mhz", hs.Summary.QuickStats.OverallCpuUsage)
-		event.Put("memory.used.bytes", int64(hs.Summary.QuickStats.OverallMemoryUsage)*1024*1024)
+		outputDsNames := make([]string, 0, len(datastores))
+		for _, ds := range datastores {
+			name := strings.ReplaceAll(ds.Name, ".", "_")
```

Comment:
Curious, what does the output of ds.Name look like that we need this replacement ?
Can you share example pelase

Reply:
I am maintaining consistency with the existing network codebase by retaining this piece of code.
I don't have an example with `.` in name.